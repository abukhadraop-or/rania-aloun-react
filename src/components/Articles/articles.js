import styled from 'styled-components';

export const Container = styled.div`
  margin: 3rem 0;
  width: 100%;
`;

export const ArticlePreview = styled.div`
  border-top: 0.13rem solid #e6e5e3;
  padding: 1.5rem 0;
`;

export const ArticleTitle = styled.p`
  color: black;
  font-size: 1.5rem;
  font-weight: 700;
  margin: 1rem 0 0;
`;

export const ArticleLink = styled.p`
  color: #bbbbbb;
  cursor: pointer;
  font-size: 0.8rem;
  margin: 0.2rem 0 0 0;
`;

export const LastLineContainer = styled.div`
  display: flex;
  justify-content: space-between;
  margin: 1rem 0 0 0;
`;

export const ReadMoreP = styled.p`
  color: #bbbbbb;
  cursor: pointer;
  font-size: 0.7rem;
  margin: 1rem 0;
`;

export const ArticleTagsContainer = styled.div`
  align-items: center;
  display: flex;
`;

export const ArticleTag = styled.p`
  border-radius: 0.8rem;
  border: 0.063rem solid #bbbbbb;
  color: #bbbbbb;
  cursor: pointer;
  font-size: 0.7rem;
  height: fit-content;
  margin: 0.125rem;
  padding: 0.2rem;
  width: fit-content;
  &:hover {
    border: 0.063rem solid #6d6f70;
  }
`;
