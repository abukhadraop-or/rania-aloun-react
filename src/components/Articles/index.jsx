import React from 'react';
import ContentNav from 'components/ContentNav';
import DisplayArticles from 'utilities/displayArticles';
import PropType from 'prop-types';
import Article from 'components/Article';

function Articles({ articles, selectedTag, currentPage }) {
  const { data } = DisplayArticles(articles, selectedTag, currentPage);
  return (
    <>
      <ContentNav />
      {data.map((article) => (
        <Article data={article} key={article.id} />
      ))}
    </>
  );
}

Articles.propTypes = {
  articles: PropType.arrayOf(
    PropType.shape({
      id: PropType.number,
    })
  ).isRequired,
  selectedTag: PropType.string.isRequired,
  currentPage: PropType.number.isRequired,
};

export default Articles;
