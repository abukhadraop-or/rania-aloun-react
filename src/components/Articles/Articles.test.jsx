import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Articles from '.';

describe('<Articles />', () => {
  it('matches snapshot', () => {
    const wrapper = shallow(
      <Articles
        articles={[
          {
            articleTitle: 'The way we look at things',
            createdAt: '2021-09-27T12:33:54.133Z',
            id: 1,
            liked: 32,
            link: 'https://css-tricks.com/snippets/css/css-box-shadow/',
            publishDate: '2021-05-08T21:00:00.000Z',
            updatedAt: '2021-10-12T10:11:00.709Z',
            userName: 'Rania Aloun',
            tags: [
              {
                TagArticle: {
                  ArticleId: 1,
                  TagId: 1,
                  createdAt: '2021-09-27T03:27:53.000Z',
                  updatedAt: '2021-09-27T03:27:53.000Z',
                },
                createdAt: '2021-09-27T03:27:53.000Z',
                id: 1,
                name: 'Article',
                updatedAt: '2021-10-12T10:11:00.709Z',
              },
            ],
          },
        ]}
        selectedTag="Article"
      />
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
