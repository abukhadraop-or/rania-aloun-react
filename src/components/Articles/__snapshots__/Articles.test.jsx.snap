// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Articles /> matches snapshot 1`] = `
<Fragment>
  <ContentNav />
  <Article
    data={
      Object {
        "articleTitle": "The way we look at things",
        "createdAt": "2021-09-27T12:33:54.133Z",
        "id": 1,
        "liked": 32,
        "link": "https://css-tricks.com/snippets/css/css-box-shadow/",
        "publishDate": "2021-05-08T21:00:00.000Z",
        "tags": Array [
          Object {
            "TagArticle": Object {
              "ArticleId": 1,
              "TagId": 1,
              "createdAt": "2021-09-27T03:27:53.000Z",
              "updatedAt": "2021-09-27T03:27:53.000Z",
            },
            "createdAt": "2021-09-27T03:27:53.000Z",
            "id": 1,
            "name": "Article",
            "updatedAt": "2021-10-12T10:11:00.709Z",
          },
        ],
        "updatedAt": "2021-10-12T10:11:00.709Z",
        "userName": "Rania Aloun",
      }
    }
    key="1"
  />
</Fragment>
`;
